### PR TITLE
Simplify handling of multi-reg returns

### LINKED ISSUE #36868: Clear/fix `compMethodReturnsMultiRegRetType` and similar functions.
On arm64 that function returns `false` for SIMD16 that is returned in one register.
`compMethodReturnsNativeScalarType` has a comment that says that it should return `true` for `SIMD16` on arm64, but it does not.

`compMethodReturnsNativeScalarType` sounds like an orthogonal method to `compMethodReturnsMultiRegRetType`, but it returns true for `LONG` on 32 bit platform when they are returned in 2 registered and `compMethodReturnsMultiRegRetType` returns `true` for them as well.

Probably the best solution would be to add `ReturnTypeDesc` into compiler as we have it for `GenTreeCall` nodes and rewrite these methods using the descriptor.

category:implementation
theme:structs
skill-level:beginner
cost:small

### PR DESCRIPTION
**Pull request by SingleAccretion**
Keep the return descriptor for the method being compiled and use it in various places.

Deletes the dependence of multi-reg `RETURN` backend code on precise handles of `LCL_VAR` operands.

Fixes #36868.

Tiny amount of [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=8468&view=ms.vss-build-web.run-extensions-tab) on ARM64: some forward substitutions of non-multireg SIMDs are now allowed.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
@dotnet/jit-contrib anyone up for reviewing this? Looks like a nice cleanup but cuts across a lot of code.

**Comment by jakobbotsch** [comment]
I can take a look, but it'll probably take me some time to get familiar with multi-reg returns.

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -10447,89 +10409,32 @@ GenTree* Compiler::impFixupStructReturnType(GenTree*                 op,
...
 
-        if (op->gtOper == GT_CALL)
+        // In contrast, we can only use multi-reg calls directly if they have the exact same ABI.
+        // Calling convention equality is a conservative approximation for that check.
+        if (op->IsCall() && (op->AsCall()->GetUnmanagedCallConv() == info.compCallConv)
+#if defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64)
+            // TODO-Review: this seems unnecessary. Return ABI doesn't change under varargs.
+            && !op->AsCall()->IsVarargs()
+#endif // defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64)
+                )
```
Comment (jakobbotsch) [inline src/coreclr/jit/importer.cpp:10419]: Seems likely this check is not necessary at all anymore after #73059.

Comment (jakobbotsch) [reply src/coreclr/jit/importer.cpp:10419]: Ah, not really.

**Review by jakobbotsch — APPROVED** [review]
This looks great to me, very nice cleanup. I'll trigger some stress legs.

**Comment by SingleAccretion** [comment]
Stress failures look like #76550, #76507 and #76280.

**Comment by jakobbotsch** [comment]
Thanks!

**Pull request merged**